estimatefee.go

```diff
@@ -0,0 +1,434 @@
+// Copyright (c) 2013-2016 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+	"sort"
+	"sync"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcutil"
+)
+
+const (
+	// estimateFeeDepth is the maximum number of blocks before a transaction
+	// is confirmed that we want to track.
+	estimateFeeBins = 25
+
+	// estimateFeeBinSize is the number of txs stored in each bin.
+	estimateFeeBinSize = 100
+
+	// estimateFeeMaxReplacements is the max number of replacements that
+	// can be made by the txs found in a given block.
+	estimateFeeMaxReplacements = 10
+)
+
+// observedTransaction represents an observed transaction and some
+// additional data required for the fee estimation algorithm.
+type observedTransaction struct {
+	hash     chainhash.Hash // a transaction hash.
```

Comment:
In order to maintain consistency with the rest of the codebase (for the most part), the comments for the struct's attributes should be placed a line `above` the attribute being described rather than `in-line`.